src/AWS.Deploy.Orchestration/Data/AWSResourceQueryer.cs

```diff
@@ -438,10 +439,44 @@ public async Task<List<Stack>> GetCloudFormationStacks()
             return await cloudFormationClient.Paginators.DescribeStacks(new DescribeStacksRequest()).Stacks.ToListAsync();
         }
 
-        public async Task<GetCallerIdentityResponse> GetCallerIdentity()
+        public async Task<GetCallerIdentityResponse> GetCallerIdentity(string awsRegion)
         {
-            using var stsClient = _awsClientFactory.GetAWSClient<IAmazonSecurityTokenService>();
-            return await stsClient.GetCallerIdentityAsync(new GetCallerIdentityRequest());
+            var request = new GetCallerIdentityRequest();
+
+            try
+            {
+                using var stsClient = _awsClientFactory.GetAWSClient<IAmazonSecurityTokenService>(awsRegion);
+                return await stsClient.GetCallerIdentityAsync(request);
+            }
+            catch (Exception ex)
+            {
+                var regionEndpoint = RegionEndpoint.GetBySystemName(awsRegion);
+                if (regionEndpoint.PartitionName.Equals("aws") && !awsRegion.Equals(Constants.CLI.DEFAULT_AWS_REGION))
+                {
+                    try
+                    {
+                        using var stsClient = _awsClientFactory.GetAWSClient<IAmazonSecurityTokenService>(Constants.CLI.DEFAULT_AWS_REGION);
+                        await stsClient.GetCallerIdentityAsync(request);
+                    }
+                    catch (Exception e)
+                    {
+                        throw new UnableToAccessAWSRegionException(
+                           DeployToolErrorCode.UnableToAccessAWSRegion,
+                           $"We were unable to access the AWS region '{awsRegion}'. Make sure you have correct permissions for that region and the region is accessible.",
+                           e);
+                    }
+
+                    throw new UnableToAccessAWSRegionException(
+                        DeployToolErrorCode.OptInRegionDisabled,
+                        $"We were unable to access the Opt-In region '{awsRegion}'. Please enable the AWS Region '{awsRegion}' and try again. Additional details could be found at https://docs.aws.amazon.com/general/latest/gr/rande-manage.html",
```

Comment:
Can you check if there is a url with an english language code? ("gr" / "rande-manage")